### DISCUSSION TITLE
Wysiwyg spell checking

### DISCUSSION THREAD

**Discussion by coes-devs-jakala** [pimcore/pimcore#17568] [Q&A / Support]
### Pimcore version

11.2.7

### Steps to reproduce

We would need to implement spell checking in wysiwyg text areas (tinyMCE).

The default editor doesn’t include it in the toolbar, although we would have expected it, ‘cause it is a fundamental feature for data-entry colleagues. Looking at the editor documentation we see that it is a paid plugin.
Is it possible to enable this plugin in the pimcore editor? Can we use a paid version of tinyMCE?

### Actual Behavior

Spell checking is not in default toolbar on wysiwyg

### Expected Behavior

Spell checking as default in wysiwyg toolbar

**Comment by NiklasBr** [comment]
You can use the operating system and browser's native spell checking by setting a [custom configuration](https://github.com/pimcore/pimcore/blob/11.x/bundles/TinymceBundle/README.md)'s value `browser_spellcheck` to `true`.

**Comment by coes-devs-jakala** [comment]
Hi @NiklasBr, this configuration enable the spellcheck of browser using the language of browser. In TinyMCE's spellcheck you can select the language used in specific field (https://www.tiny.cloud/docs/tinymce/latest/introduction-to-tiny-spellchecker/). How can i configurate this plugin?

**Comment by NiklasBr** [comment]
No additional configuration of the server/TinyMCE needed other than the setting above.

Some guides for common browsers:

[How do I use the Firefox spell checker?](https://support.mozilla.org/en-US/kb/how-do-i-use-firefox-spell-checker)

[Turn Chrome spell check on and off](https://support.google.com/chrome/answer/12027911?hl=en)

[How using Microsoft Edge can improve your writing and grammar](https://www.microsoft.com/en-us/edge/learning-center/improve-writing-and-grammar?form=MA13I2)

**Comment by coes-devs-jakala** [comment]
Thanks for the suggestion, but is there a way to install plugin premium of TinyMCE or not? If yes, how?

**Comment by NiklasBr** [comment]
You [first need](https://www.tiny.cloud/docs/tinymce/latest/introduction-to-premium-selfhosted-services/) TinyMCE cloud or "Server-side components must be deployed onto a Java Servlet 3.0 compatible application server" and Pimcore does not use either.

**Comment by kingjia90** [comment]
You just have to subscribe to `Professional` subscription (NB: the "lower" tiers do not provide that) and get a license 
![image](https://github.com/user-attachments/assets/51f9cc7f-11ac-4e88-87df-0c194eb552b3)
source:
https://www.tiny.cloud/pricing/ (check there if prices changes over time and the screenshot gets outdated)

Then you have to add this entry [license_key](https://www.tiny.cloud/docs/tinymce/latest/license-key/#use-tinymce-with-a-commercial-license) to the global configs
https://github.com/pimcore/pimcore/blob/5b0399603ac007f827fcb97f522ab1f592acfb1b/bundles/TinymceBundle/README.md?plain=1#L75-L76
which gets then [assigned](https://github.com/pimcore/pimcore/blob/7ddfcbeefd176fee16c1e8a87a5acdc3176aec29/bundles/TinymceBundle/public/js/editor.js#L150) to the `tinymce.init`

Then follow the instruction to enable the plugin and customize its configuration
https://www.tiny.cloud/docs/tinymce/latest/introduction-to-tiny-spellchecker/


Mhhh, nothing to do here on Pimcore side, other than maybe put some placeholder config and improve documentation

**Reply by kingjia90** [reply]
@coes-devs-jakala i've converted the issue to a discussion entry as it couldn't be considered as a bug.
If you need anything, please continue the discussion here and if it works, please also mark my comment above `as answer` and share eventually any blockers or pro tips you may encounter when setting it up, so that it may help others too. Thank you in advance!
